aptible/resource_database.go

```diff
@@ -163,45 +241,81 @@ func resourceDatabaseImport(d *schema.ResourceData, meta interface{}) ([]*schema
+	legacy := meta.(*providerMetadata).LegacyClient
+	databaseID := int32(d.Get("database_id").(int))
+	containerSize := int32(d.Get("container_size").(int))
+	profile := d.Get("container_profile").(string)
+	diskSize := int32(d.Get("disk_size").(int))
+	iops := int32(d.Get("iops").(int))
 	handle := d.Get("handle").(string)
 	var diags diag.Diagnostics
 
-	updates := aptible.DBUpdates{}
+	ctx = meta.(*providerMetadata).APIContext(ctx)
+	payload := aptibleapi.NewCreateOperationRequest("restart")
 
 	if d.HasChange("container_size") {
-		updates.ContainerSize = containerSize
+		payload.SetContainerSize(containerSize)
+	}
+	if d.HasChange("iops") {
+		payload.SetProvisionedIops(iops)
 	}
 	if d.HasChange("container_profile") {
-		updates.ContainerProfile = containerProfile
+		payload.SetInstanceProfile(profile)
 	}
 	if d.HasChange("disk_size") {
-		updates.DiskSize = diskSize
+		payload.SetDiskSize(diskSize)
 	}
 
 	if d.HasChange("handle") {
-		log.Printf("[INFO] Updating handle to %s\n", handle)
-		updates.Handle = handle
+		_, err := client.
+			DatabasesAPI.
+			PatchDatabase(ctx, databaseID).
+			UpdateDatabaseRequest(
+				aptibleapi.UpdateDatabaseRequest{Handle: &handle},
+			).
+			Execute()
+		if err != nil {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "There was an error when trying to update the database handle.",
+				Detail:   generateErrorFromClientError(err).Error(),
+			})
+			return diags
+		}
 	}
 
-	// if only changing the handle, you can skip running an operation needlessly
-	// below can be hard to read, but it basically means if nothing besides handle was changed
-	if !d.HasChangesExcept("handle") {
-		updates.SkipOperationUpdate = true
-	}
+	if d.HasChangeExcept("handle") {
+		op, _, err := client.
+			OperationsAPI.
+			CreateOperationForDatabase(ctx, databaseID).
+			CreateOperationRequest(*payload).Execute()
+		if err != nil {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "There was an error when trying to update the database.",
+				Detail:   generateErrorFromClientError(err).Error(),
+			})
+			return diags
+		}
 
-	err := client.UpdateDatabase(databaseID, updates)
-	if err != nil {
-		diags = append(diags, diag.Diagnostic{
-			Severity: diag.Error,
-			Summary:  "There was an error when trying to update the database.",
-			Detail:   generateErrorFromClientError(err).Error(),
-		})
-		return diags
+		del, err := legacy.WaitForOperation(int64(op.Id))
+		if err != nil {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "There was an error when trying to update the database.",
+				Detail:   generateErrorFromClientError(err).Error(),
+			})
+			return diags
+		}
+		if del {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  fmt.Sprintf("The replica with handle: %s was unexpectedly deleted", handle),
+				Detail:   generateErrorFromClientError(err).Error(),
```

Comment:
`err` is `nil` at this point. I recommend removing the detail or changing the summary e.g.

```
Summary: fmt.Sprintf( "Error updating replica with handle: %s, handle),
Detail:  "The replica was unexpectedly deleted",
```